### PR TITLE
fixed the broken path in apx

### PR DESCRIPTION
**Pull request by 9glenda**
changed `/etc/apx/` to the config in the nix store to make apx work without setting up a config in /etc

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by dit7ya — CHANGES_REQUESTED** [review]
Please change the commit message to `apx: patch config file path`.

File: pkgs/tools/package-management/apx/default.nix [hunk 1]
```diff
@@ -26,6 +26,9 @@ buildGoModule rec {
     makeWrapper
     installShellFiles
   ];
+  patchPhase = ''
```
Comment (SuperSandro2000) [inline pkgs/tools/package-management/apx/default.nix:29]:
```suggestion
  ];

  postPatch = ''
```

**Comment by winterqt** [comment]
> the PR should be squashed merged and there its possible to change to commit message right? I could also force push it if you want to.

Please squash manually, per [the contributing guide](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#writing-good-commit-messages).

**Comment by dit7ya** [comment]
@9glenda Does this patch allow for editing the configuration file at all after installation? If not I this creates more restriction on the users as the configuration file cannot be customized?

**Comment by yswtrue** [comment]
please merge this pr

**Pull request closed without merging**